pkg/osvscanner/osvscanner.go

```diff
@@ -357,29 +357,75 @@ func scanDebianDocker(r *output.Reporter, query *osv.BatchedQuery, dockerImageNa
 	return nil
 }
 
-// Filters response according to config, returns number of responses removed
-func filterResponse(r *output.Reporter, query osv.BatchedQuery, resp *osv.BatchedResponse, configManager *config.ConfigManager) int {
-	hiddenVulns := map[string]config.IgnoreEntry{}
-
-	for i, result := range resp.Results {
-		var filteredVulns []osv.MinimalVulnerability
-		configToUse := configManager.Get(r, query.Queries[i].Source.Path)
-		for _, vuln := range result.Vulns {
-			ignore, ignoreLine := configToUse.ShouldIgnore(vuln.ID)
-			if ignore {
-				hiddenVulns[vuln.ID] = ignoreLine
-			} else {
-				filteredVulns = append(filteredVulns, vuln)
+// Filters results according to config, preserving order. Returns total number of vulnerabilities removed.
+func filterResults(r *output.Reporter, results *models.VulnerabilityResults, configManager *config.ConfigManager) int {
+	removedCount := 0
+	newResults := []models.PackageSource{}
+	for _, pkgSrc := range results.Results {
+		configToUse := configManager.Get(r, pkgSrc.Source.Path)
+		newPackages := []models.PackageVulns{}
+		for _, pkgVulns := range pkgSrc.Packages {
+			newVulns := filterPackageVulns(r, pkgVulns, configToUse)
+			removedCount += len(pkgVulns.Vulnerabilities) - len(newVulns.Vulnerabilities)
+			if len(newVulns.Vulnerabilities) > 0 {
+				newPackages = append(newPackages, newVulns)
 			}
 		}
-		resp.Results[i].Vulns = filteredVulns
+		if len(newPackages) > 0 {
+			pkgSrc.Packages = newPackages
+			newResults = append(newResults, pkgSrc)
+		}
+	}
+	results.Results = newResults
+
+	return removedCount
+}
+
+// Filters package-grouped vulnerabilities according to config, preserving ordering. Returns filtered package vulnerabilities.
+func filterPackageVulns(r *output.Reporter, pkgVulns models.PackageVulns, configToUse config.Config) models.PackageVulns {
+	hiddenVulns := map[string]bool{}
```

Comment:
nit: we don't actually need this to have a bool value. We can just use `map[string]struct{}` which is slightly more efficient (0 byte storage) and a common pattern.